src/main/java/frc/robot/subsystems/localization/Localizer.java

```diff
@@ -0,0 +1,104 @@
+package frc.robot.subsystems.localization;
+
+import edu.wpi.first.apriltag.*;
+import edu.wpi.first.cameraserver.CameraServer;
+import edu.wpi.first.cscore.CvSink;
+import edu.wpi.first.cscore.CvSource;
+import edu.wpi.first.cscore.UsbCamera;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Constants;
+import java.util.ArrayList;
+import org.opencv.core.Mat;
+import org.opencv.core.Point;
+import org.opencv.core.Scalar;
+import org.opencv.imgproc.Imgproc;
+
+public class Localizer {
+  private Thread visionThread;
+  private AprilTagDetector detector;
+
+  public Localizer() {
+    detector = new AprilTagDetector();
+    detector.addFamily("tag16h5", 0);
+    AprilTagDetector.Config config = new AprilTagDetector.Config();
+    detector.setConfig(config);
+
+    visionThread =
+        new Thread(
+            () -> {
+              // Get the UsbCamera from CameraServer
+              UsbCamera camera = CameraServer.startAutomaticCapture();
+
+              // Set the resolution
+              camera.setResolution(640, 480);
+
+              // Get a CvSink. This will capture Mats from the camera
+              CvSink cvSink = CameraServer.getVideo();
+
+              // Setup a CvSource. This will send images back to the Dashboard
+              CvSource outputStream = CameraServer.putVideo("Rectangle", 640, 480);
+
+              // Mats are very memory expensive. Lets reuse this Mat.
+              Mat mat = new Mat();
+              Mat grayMat = new Mat();
+              ArrayList<Integer> tags = new ArrayList<>();
+              AprilTagPoseEstimator.Config poseEstConfig =
+                  new AprilTagPoseEstimator.Config(
+                      Constants.TARGET_SIZE_METERS,
```

Comment:
I notice these lines are being called from Constants class, but no values are located in that class